### PR TITLE
Improve doc snippet test coverage in snippets/kotlinDsl

### PR DESCRIPTION
**Pull request by mlopatkin**
Added test coverage, no changes to the userguide.

### CODE REVIEW

**Comment by mlopatkin** [comment]
@bot-gradle test this

**Comment by mlopatkin** [comment]
@bot-gradle test this

**Review by 6hundreds — APPROVED** [review]
LGTM!

**Comment by mlopatkin** [comment]
@bot-gradle test and merge

File: subprojects/docs/build.gradle [hunk 1]
```diff
@@ -631,6 +631,8 @@ tasks.named("docsTest") { task ->
             excludeTestsMatching "org.gradle.docs.samples.*.structuring-software-projects*android-app.sample"
             // Umbrella build project contains also Android projects so it requires Java 11+
             excludeTestsMatching "org.gradle.docs.samples.*.structuring-software-projects_*_umbrella-build.sample"
+            // This test sets source and target compatibility to 11
```
Comment (bamboo) [inline subprojects/docs/build.gradle:634]: 👍

**Review by bamboo — COMMENTED** [review]
👍

**Pull request merged**